Description: Fix multi-page HTML manual
 Two new chapters were annotated with the same HTML filename and so overwrite
 one another.  Fixed upstream already.
Author: Olly Betts <olly@survex.com>
Forwarded: not-needed
Last-Update: 2024-07-05

--- a/doc/manual/future.htm	2024-07-05 08:23:52.136169377 +1200
+++ b/doc/manual/future.htm	2024-07-05 08:24:38.960212777 +1200
@@ -11,7 +11,7 @@
 HREF="index.htm"><LINK
 REL="PREVIOUS"
 TITLE="Working with Bob Thrun's CMAP"
-HREF="walls.htm"></HEAD
+HREF="cmap.htm"></HEAD
 ><BODY
 CLASS="SECT1"
 ><DIV
@@ -37,7 +37,7 @@
 ALIGN="left"
 VALIGN="bottom"
 ><A
-HREF="walls.htm"
+HREF="cmap.htm"
 ACCESSKEY="P"
 >Prev</A
 ></TD
@@ -102,7 +102,7 @@
 ALIGN="left"
 VALIGN="top"
 ><A
-HREF="walls.htm"
+HREF="cmap.htm"
 ACCESSKEY="P"
 >Prev</A
 ></TD
--- a/doc/manual/index.htm	2024-07-05 08:23:52.136169377 +1200
+++ b/doc/manual/index.htm	2024-07-05 08:24:38.960212777 +1200
@@ -155,7 +155,7 @@
 ></DT
 ><DT
 ><A
-HREF="walls.htm"
+HREF="cmap.htm"
 >Working with Bob Thrun's CMAP</A
 ></DT
 ><DT
--- a/doc/manual/walls.htm	2024-07-05 08:23:52.132169373 +1200
+++ b/doc/manual/walls.htm	2024-07-05 08:24:38.948212767 +1200
@@ -2,7 +2,7 @@
 <HTML
 ><HEAD
 ><TITLE
->Working with Bob Thrun's CMAP</TITLE
+>Working with David McKenzie's Walls</TITLE
 ><META
 NAME="GENERATOR"
 CONTENT="Modular DocBook HTML Stylesheet Version 1.79"><LINK
@@ -10,11 +10,11 @@
 TITLE="Survex 1.4.9 Manual"
 HREF="index.htm"><LINK
 REL="PREVIOUS"
-TITLE="Working with David McKenzie's Walls"
-HREF="walls.htm"><LINK
+TITLE="Mailing List"
+HREF="maillist.htm"><LINK
 REL="NEXT"
-TITLE="Future Developments"
-HREF="future.htm"></HEAD
+TITLE="Working with Bob Thrun's CMAP"
+HREF="cmap.htm"></HEAD
 ><BODY
 CLASS="SECT1"
 ><DIV
@@ -40,7 +40,7 @@
 ALIGN="left"
 VALIGN="bottom"
 ><A
-HREF="walls.htm"
+HREF="maillist.htm"
 ACCESSKEY="P"
 >Prev</A
 ></TD
@@ -54,7 +54,7 @@
 ALIGN="right"
 VALIGN="bottom"
 ><A
-HREF="future.htm"
+HREF="cmap.htm"
 ACCESSKEY="N"
 >Next</A
 ></TD
@@ -68,41 +68,179 @@
 ><H1
 CLASS="SECT1"
 ><A
-NAME="AEN1955"
->Working with Bob Thrun's CMAP</A
+NAME="AEN1905"
+>Working with David McKenzie's Walls</A
 ></H1
 ><P
->Survex can read CMAP processed survey data, commonly known as CMAP XYZ files.
-CMAP no longer seems to be used, but we've kept the support in place so
-it's there if anyone finds old data and wants to view it.</P
+>Survex 1.4.9 and later can read Walls unprocessed survey data (.SRV and .WPJ
+files).  The support is currently somewhat experimental, as the documentation
+of the SRV format is incomplete and incorrect in places, while the WPJ format
+seems to be largely undocumented, and David is sadly no longer around to ask.</P
 ><P
->Support was added long ago (Survex 1.0.4) but you really want to use Survex
-1.4.9 or later due to a feet to metres conversion bug in all versions before
-this, which scaled all returned coordinates from CMAP XYZ files by a factor of
-about 10.76.</P
+>The current status is that most .SRV files should process individually (but
+see details below of features which are not handled), while there are still some
+known issues which cause problems with some .WPJ files.</P
 ><P
->CMAP XYZ files contain a timestamp.  CMAP was originally written for computers
-where the clock was just set to localtime so it seems likely this timestamp is
-in localtime, but it does not specify a timezone. Survex assumes it's in UTC,
-which is at least fairly central in the possibilities, but may mean timestamps
-are off by up to about half a day. The timestamps in example files all use two
-digit years. It's not documented how CMAP handles years 2000 and later, so we
-assume 4 digit years are correct, years &lt; 50 are 20xx, and other years get
-1900 added to them (so year 101 is 2001).</P
+>Walls is no longer being developed, so the focus of support for Walls
+formats is primarily to help people with existing Walls data to migrate.
+If you are in this situation and the incomplete WPJ support is a problem, you
+should be able to write a .svx file to replace your WPJ file - you can use
+"*include somedata.srv" to include a Walls .srv from a .svx file.</P
 ><P
->CMAP XYZ files don't seem to contain any stations flags.  There's a single
-character "type" which acts as a leg flag of sorts, but it seems the
-meaning is up to the user so we don't try to interpret it.  We assume all
-the data is underground (so all stations get the "underground" flag).</P
 ><P
->There are two variants of CMAP XYZ files.  CMAP v16 and later produce
-the "shot" variant (with extension ".sht"), which is well supported by
-Survex.</P
+></P
+><UL
+><LI
 ><P
->Older CMAP versions produced the "station" variant (with extension ".adj" for
-adjusted data and ".una" for unadjusted data).  Survex only supports
-reading stations from these - the survey legs linking them are currently
-ignored.</P
+>	Walls allows hanging surveys, whereas these are currently treated as an
+	error by Survex.  This is probably the current top priority to address.
+    </P
+></LI
+><LI
+><P
+>	Survex reports warnings in some suspect situations which Walls quietly
+	accepts.  In general this seems helpful, but if there are particular
+	instances which are noisy and not useful, let us know.
+    </P
+></LI
+><LI
+><P
+>	#FIX - currently Survex does not support horizontal-only or vertical
+	only fixes. These are currently given an SD of 1000m in that direction
+	instead, but that is not the same as completely decoupling the fix in
+	that direction.
+    </P
+></LI
+><LI
+><P
+>	#FIX - degree:minute:second fixes (e.g. W97:43:52.5) are not currently
+	supported.
+    </P
+></LI
+><LI
+><P
+>	Walls FLAG values seems to be arbitrary text strings.  We try to infer
+	appropriate Survex station flags by checking for certain key words in
+	that text (currently we map words ENTRANCE and FIX to the corresponding
+	Survex station flags) and otherwise ignore FLAG values.
+    </P
+></LI
+><LI
+><P
+>	We don't currently support all the datum names which Walls does because
+	we haven't managed to find an EPSG code for any UTM zones in some of
+	these datums.  This probably means they're not actually in current
+	use.
+    </P
+></LI
+><LI
+><P
+>	We currently assume all two digit years are 19xx (Walls documents
+	it 'also accepts "some date formats common in the U.S. (mm/dd/yy,
+	mm-dd-yyyy, etc.)' but doesn't say how it interprets 'yy'.
+    </P
+></LI
+><LI
+><P
+>	The documentation specifies that the SAVE and RESTORE options should be
+	processed before other options.  Currently Survex just processes all
+	options in the order specified, which makes no difference to any
+	real-world data we've seen. We need to test with Walls32.exe to
+	determine exactly how this works (and if RESET is also special).
+    </P
+></LI
+><LI
+><P
+>	LRUD data is currently ignored.
+    </P
+></LI
+><LI
+><P
+>	The TAPE= option is currently quietly skipped, and tape measurements
+	are assumed to be station to station.
+    </P
+></LI
+><LI
+><P
+>	In TYPEAB= and TYPEVB=, the threshold is ignored, as is the X meaning to
+	only use foresights (but still check backsights). Survex uses a
+	threshold based on the specified instrument SDs, and averages foresights
+	and backsights.
+    </P
+></LI
+><LI
+><P
+>	FLAG= is quietly skipped.
+    </P
+></LI
+><LI
+><P
+>	UV=, UVH= and UVV= are all quietly skipped.
+    </P
+></LI
+><LI
+><P
+>	The GRID= option currently gives an "Unknown command" warning, and is
+	skipped.  If your Walls data specifies a UTM zone then Survex will
+	automatically correct for grid convergence.
+    </P
+></LI
+><LI
+><P
+>	The INCH= option currently gives an "Unknown command" warning, and is
+	skipped.
+    </P
+></LI
+><LI
+><P
+>	The RECT= option currently gives an "Unknown command" warning, and is
+	skipped (this option specifies how much to orient cartesian style data
+	relative to true North, not to be confused with the RECT option which
+	is upported).
+    </P
+></LI
+><LI
+><P
+>	Walls documents allowing a "maxiumum of eight characters" in unprefixed
+	names - we don't bother trying to enforce this restriction, but this
+	should not make a difference in valid data.
+    </P
+></LI
+><LI
+><P
+>	Walls seems to allow \ in place of / in some places (e.g. #FLAG).  We
+	aim to support this too, but it doesn't seem to be documented so may
+	not currently be supported in the correct places.
+    </P
+></LI
+><LI
+><P
+>	The inheritance of settings in WPJ files is probably not correctly
+	implemented currently.
+    </P
+></LI
+><LI
+><P
+>	The Walls documentation mentions a NOTE= option, but doesn't document
+	what it does, and testing with Walls32.exe it doesn't seem to actually
+	be supported!
+    </P
+></LI
+><LI
+><P
+>	The two UPS zones for the polar regions (specified as UTM zone values
+	of -61 and 61 in Walls) are supported with datum WGS84, but we do not
+	have any real data to test this support with.
+    </P
+></LI
+></UL
+></P
+><P
+>If you find some Walls data which Survex doesn't handle or handles incorrectly,
+and it is not already noted above, please let us know. If you can provide some
+data demonstrating the problem, that's really helpful. It's also useful to know
+if there are things listed above that are problematic to help prioritise
+efforts.</P
 ></DIV
 ><DIV
 CLASS="NAVFOOTER"
@@ -120,7 +258,7 @@
 ALIGN="left"
 VALIGN="top"
 ><A
-HREF="walls.htm"
+HREF="maillist.htm"
 ACCESSKEY="P"
 >Prev</A
 ></TD
@@ -138,7 +276,7 @@
 ALIGN="right"
 VALIGN="top"
 ><A
-HREF="future.htm"
+HREF="cmap.htm"
 ACCESSKEY="N"
 >Next</A
 ></TD
@@ -148,7 +286,7 @@
 WIDTH="33%"
 ALIGN="left"
 VALIGN="top"
->Working with David McKenzie's Walls</TD
+>Mailing List</TD
 ><TD
 WIDTH="34%"
 ALIGN="center"
@@ -158,7 +296,7 @@
 WIDTH="33%"
 ALIGN="right"
 VALIGN="top"
->Future Developments</TD
+>Working with Bob Thrun's CMAP</TD
 ></TR
 ></TABLE
 ></DIV
--- a/doc/manual/cmap.htm	1970-01-01 12:00:00.000000000 +1200
+++ b/doc/manual/cmap.htm	2024-07-05 08:24:38.956212774 +1200
@@ -0,0 +1,167 @@
+<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN""http://www.w3.org/TR/html4/loose.dtd">
+<HTML
+><HEAD
+><TITLE
+>Working with Bob Thrun's CMAP</TITLE
+><META
+NAME="GENERATOR"
+CONTENT="Modular DocBook HTML Stylesheet Version 1.79"><LINK
+REL="HOME"
+TITLE="Survex 1.4.9 Manual"
+HREF="index.htm"><LINK
+REL="PREVIOUS"
+TITLE="Working with David McKenzie's Walls"
+HREF="walls.htm"><LINK
+REL="NEXT"
+TITLE="Future Developments"
+HREF="future.htm"></HEAD
+><BODY
+CLASS="SECT1"
+><DIV
+CLASS="NAVHEADER"
+><TABLE
+SUMMARY="Header navigation table"
+WIDTH="100%"
+BORDER="0"
+CELLPADDING="0"
+CELLSPACING="0"
+><TR
+><TH
+COLSPAN="3"
+ALIGN="center"
+><SPAN
+CLASS="APPLICATION"
+>Survex</SPAN
+> 1.4.9 Manual</TH
+></TR
+><TR
+><TD
+WIDTH="10%"
+ALIGN="left"
+VALIGN="bottom"
+><A
+HREF="walls.htm"
+ACCESSKEY="P"
+>Prev</A
+></TD
+><TD
+WIDTH="80%"
+ALIGN="center"
+VALIGN="bottom"
+></TD
+><TD
+WIDTH="10%"
+ALIGN="right"
+VALIGN="bottom"
+><A
+HREF="future.htm"
+ACCESSKEY="N"
+>Next</A
+></TD
+></TR
+></TABLE
+><HR
+ALIGN="LEFT"
+WIDTH="100%"></DIV
+><DIV
+CLASS="SECT1"
+><H1
+CLASS="SECT1"
+><A
+NAME="AEN1955"
+>Working with Bob Thrun's CMAP</A
+></H1
+><P
+>Survex can read CMAP processed survey data, commonly known as CMAP XYZ files.
+CMAP no longer seems to be used, but we've kept the support in place so
+it's there if anyone finds old data and wants to view it.</P
+><P
+>Support was added long ago (Survex 1.0.4) but you really want to use Survex
+1.4.9 or later due to a feet to metres conversion bug in all versions before
+this, which scaled all returned coordinates from CMAP XYZ files by a factor of
+about 10.76.</P
+><P
+>CMAP XYZ files contain a timestamp.  CMAP was originally written for computers
+where the clock was just set to localtime so it seems likely this timestamp is
+in localtime, but it does not specify a timezone. Survex assumes it's in UTC,
+which is at least fairly central in the possibilities, but may mean timestamps
+are off by up to about half a day. The timestamps in example files all use two
+digit years. It's not documented how CMAP handles years 2000 and later, so we
+assume 4 digit years are correct, years &lt; 50 are 20xx, and other years get
+1900 added to them (so year 101 is 2001).</P
+><P
+>CMAP XYZ files don't seem to contain any stations flags.  There's a single
+character "type" which acts as a leg flag of sorts, but it seems the
+meaning is up to the user so we don't try to interpret it.  We assume all
+the data is underground (so all stations get the "underground" flag).</P
+><P
+>There are two variants of CMAP XYZ files.  CMAP v16 and later produce
+the "shot" variant (with extension ".sht"), which is well supported by
+Survex.</P
+><P
+>Older CMAP versions produced the "station" variant (with extension ".adj" for
+adjusted data and ".una" for unadjusted data).  Survex only supports
+reading stations from these - the survey legs linking them are currently
+ignored.</P
+></DIV
+><DIV
+CLASS="NAVFOOTER"
+><HR
+ALIGN="LEFT"
+WIDTH="100%"><TABLE
+SUMMARY="Footer navigation table"
+WIDTH="100%"
+BORDER="0"
+CELLPADDING="0"
+CELLSPACING="0"
+><TR
+><TD
+WIDTH="33%"
+ALIGN="left"
+VALIGN="top"
+><A
+HREF="walls.htm"
+ACCESSKEY="P"
+>Prev</A
+></TD
+><TD
+WIDTH="34%"
+ALIGN="center"
+VALIGN="top"
+><A
+HREF="index.htm"
+ACCESSKEY="H"
+>Home</A
+></TD
+><TD
+WIDTH="33%"
+ALIGN="right"
+VALIGN="top"
+><A
+HREF="future.htm"
+ACCESSKEY="N"
+>Next</A
+></TD
+></TR
+><TR
+><TD
+WIDTH="33%"
+ALIGN="left"
+VALIGN="top"
+>Working with David McKenzie's Walls</TD
+><TD
+WIDTH="34%"
+ALIGN="center"
+VALIGN="top"
+>&nbsp;</TD
+><TD
+WIDTH="33%"
+ALIGN="right"
+VALIGN="top"
+>Future Developments</TD
+></TR
+></TABLE
+></DIV
+></BODY
+></HTML
+>
\ No newline at end of file
